controller/src/main/java/org/jboss/as/controller/logging/ControllerLogger.java

```diff
@@ -3505,6 +3505,9 @@ OperationFailedRuntimeException capabilityAlreadyRegisteredInContext(String capa
             "This is not valid for a 'profile' resource on process type %s so this step will not be executed.")
     void invalidRuntimeStageForProfile(String operation, String address, OperationContext.Stage stage, ProcessType processType);
 
-    @Message(id = 4445, value = "%s with value '%s' in attribute %s is already defined")
+    @Message(id = 445, value = "%s with value '%s' in attribute %s is already defined")
     OperationFailedException alreadyDefinedAttribute(String resource, String value, String attribute);
+
+    @Message(id = 446, value = "%s or alternative(s) %s is required")
+    String requiredWithAlternatives(String name, Set<String> alternatives);
```

Comment:
This can return OperationFailedException directly and constructing it in ValidateModelStepHandler can be skipped.  The existing method can be changed too:

```
    @Message(id = 172, value = "%s is required")
    OperationFailedException required(String name);
```